### PR TITLE
Post Featured Image Block: Border radius and color not applied on overlay

### LINKED ISSUE #44262: Border radius is not applied to overlay element
### Description

If the Post Featured Image or Cover block has a border-radius set in theme.json, and the block has an overlay set, the border-radius is not applied to the overlay element.

With the Post Featured image block, the overlay shows with square corners on top of the border-radius:
image

For the featured image, I think we should apply the border-radius to the overlay, as well as the image.

With the Cover block, the border-radius does not appear at all (it's applied to the container element):
image

For the cover block, it looks like the border-radius needs to be applied to the nested image and the overlay.

This issue was found when testing the Canary style variation in Twenty Twenty-Three.

### Step-by-step reproduction instructions

1. Activate the Twenty Twenty-Three theme and select the Canary variation
2. Open the canary.json file in styles directory and add the following under `styles.blocks`:
```
"core/cover": {
	"border": {
		"radius": "100px 0 0 0"
	}
},
```
3. Notice that the border-radius is not applied correctly to the overlay on the front end

To test the featured image, use the following markup in place of the cover block in the single template:

```
<!-- wp:post-featured-image {"overlayColor":"contrast","dimRatio":50,"align":"wide"} /-->
```

The border-radius for the featured image is already set in canary.json, under `core/post-featured-image`. Notice how this is not applied correctly.

### Screenshots, screen recording, code snippet

_No response_

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by RahiDroid**
## What?
- Fixes border-radius not being applied to the overlay when added from the `theme.json` file. 
- Fixes border-radius not being applied to the overlay when different values are added for each corner/side.
- Fixes border-color not being applied on the overlay. 

## Why?
- Partially covers https://github.com/WordPress/gutenberg/issues/44262

## How?
- Uses the style engine-generated attributes for the border and passes it on to the Overlay element.

## Screenshots or screencast
| Before | After |
|--------|------|
|Screenshot 2022-09-20 at 6 47 09 PM|Screenshot 2022-09-20 at 6 44 41 PM|

### CODE REVIEW

File: packages/block-library/src/post-featured-image/block.json [hunk 1]
```diff
@@ -62,7 +62,7 @@
 			"color": true,
 			"radius": true,
 			"width": true,
-			"__experimentalSelector": "img, .block-editor-media-placeholder",
+			"__experimentalSelector": "img, .block-editor-media-placeholder, .wp-block-post-featured-image__overlay",
```
Comment (mikachan) [inline packages/block-library/src/post-featured-image/block.json:65]: Apologies that I missed this PR - I see https://github.com/WordPress/gutenberg/pull/44286 is a duplicate of the fix on this line.

**Review by mikachan — APPROVED** [review]
Thanks for this! It's working well for me, and I like the way you've used `get_block_core_post_featured_image_border_attributes` for the border styles.

File: packages/block-library/src/post-featured-image/block.json (hunk 1 above)
Comment (RahiDroid) [reply packages/block-library/src/post-featured-image/block.json:65]: Thank you for reviewing Sarah!

Yes, this PR was in the draft, probably that's why Aaron raised that quick fix to help us unblock the ticket you had created.

**Comment by aaronrobertshaw** [comment]
Thanks for working on this @RahiDroid 👍 

As noted in my [comment](https://github.com/WordPress/gutenberg/issues/44262#issuecomment-1253205062) on https://github.com/WordPress/gutenberg/issues/44262, we've got some overlap between our PRs addressing these overlay issues.

https://github.com/WordPress/gutenberg/pull/44286 also addresses a problem where the markup relating to the overlay is rendered in the editor when it isn't on the frontend.

If it is ok with you, I'd like to close this PR in favour of https://github.com/WordPress/gutenberg/pull/44286. What do you think?

**Comment by aaronrobertshaw** [comment]
As [discussed](https://github.com/WordPress/gutenberg/issues/44262#issuecomment-1253614165), closing this one in favour of https://github.com/WordPress/gutenberg/pull/44286

_Note: I see this had another commit pushed after ok to close it however it missed adding the border classes to the overlay in the editor as the other PR does. So we'll still proceed with #44286_

**Pull request closed without merging**